src/plugins/discover/public/application/view_components/canvas/discover_table.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import { History } from 'history';
+import { IndexPattern } from '../../../opensearch_dashboards_services';
+import { DiscoverServices } from '../../../build_services';
+import { SavedSearch } from '../../../saved_searches';
+import { DiscoverTableService } from './discover_table_service';
+import { fetchIndexPattern, fetchSavedSearch } from '../utils/index_pattern_helper';
+
+export interface DiscoverTableProps {
+  services: DiscoverServices;
+  history: History;
+}
+
+export const DiscoverTable = ({ history, services }: DiscoverTableProps) => {
+  const { core, chrome, data, uiSettings: config, toastNotifications } = services;
+  const [savedSearch, setSavedSearch] = useState<SavedSearch>();
+  const [indexPattern, setIndexPattern] = useState<IndexPattern | undefined>(undefined);
+  // ToDo: get id from data explorer since it is handling the routing logic
```

Comment:
nit: I usually have my case sensitivity turned on so i dont want to accidentally miss this
```suggestion
  // TODO: get id from data explorer since it is handling the routing logic
```

Reply:
will replace all ToDo to TODO